prime-angular-frontend/src/app/modules/paper-enrolment/shared/guards/paper-enrolment.guard.ts

```diff
@@ -26,29 +27,40 @@ export class PaperEnrolmentGuard extends BaseGuard {
   }
 
   protected checkAccess(routePath: string = null, params: Params): Observable<boolean> | Promise<boolean> {
-    const enrolleeId = params.eid;
-    if (!enrolleeId) {
+    const enrolleeId = +params.eid;
+    if (enrolleeId) {
       return this.paperEnrolmentResource.getEnrolleeById(enrolleeId)
         .pipe(
-          map((enrollee: HttpEnrollee) => {
-            // Store the enrollee for access throughout creation and updating of a
-            // site, which will allows provide the most up-to-date site
-            // TODO replace each enrollee GET for the service
-            // this.paperEnrolmentService.enrollee = enrollee;
-
-            return this.routeDestination(routePath, enrollee);
-          })
+          map((enrollee: HttpEnrollee) => this.routeDestination(routePath, enrollee, params))
         );
-    } else {
-      return new Promise(async (resolve, reject) => resolve(true));
     }
+    return of(true);
   }
 
   /**
    * @description
    * Determine the route destination based on the enrolment.
    */
-  private routeDestination(routePath: string, enrolment: HttpEnrollee) {
-    return !!enrolment;
+  private routeDestination(routePath: string, httpEnrollee: HttpEnrollee, params: Params) {
+    if (!httpEnrollee?.approvedDate) {
```

Comment:
You can do the same thing with a ternary in this method as well, or use null coalesce:

```
return (httpEnrollee?.approvedDate)
  ? this.navigate(...)
  : true;
```

```
return !httpEnrollee?.approvedDate ?? this.navigate(...);
```